### ISSUE TITLE
Runtime Update

### ISSUE THREAD

**Issue by Justinzobel** [flathub/org.openclonk.OpenClonk#8]
Please consider upgrading to a Qt6 runtime if your application supports Qt6, or 5.15-2308 otherwise.

**Comment by jcaesar** [comment]
Any specific reasons for that? Especially, is `org.kde.Platform` going to be EOL soon?

**Comment by Justinzobel** [comment]
It will eventually stop receiving updates. Which means security issues not getting patched, newer features not being available, etc. Just like your OS it's a good idea to keep it up to date.

**Comment by Justinzobel** [comment]
Actually, the 21.08 runtime is already EOL (End Of Life). This message appears when users install the app.

**Comment by jcaesar** [comment]
> It will eventually stop receiving updates.

That's true for the most recent version also. This thing here is very much in maintainance mode, so I'd like to not spend time on frequent updates. Sorry. ^^

> Actually, the 21.08 runtime is already EOL

That sounds like a different problem from the initial issue report, but good to know. I'll see if it can be done easily or whether I need a deeper look some time in the future.

**Comment by jcaesar** [comment]
Finally got around to this, and it seems to build and play fine with both 5.15-24.08 or 6.8. Is there a reason to pick one over the other, especially if I don't want to have to look at it too often? (With a little bit of effort, I might also be able to migrate this to freedesktop. I suppose it wouldn't have a different release cycle than 5.15?)

(Also, I misunderstood you initial message about a Qt6 runtime. You just meant org.kde.Platform 6.x, right?)

**Comment by Justinzobel** [comment]
5.15-24.08 is the Qt5 branch.
6.8 is the Qt6 branch.

If it works on Qt6, use Qt6. Qt5 will eventually stop being supported.

**Comment by lluchs** [comment]
For now, only Qt 5 will work. I have [a patch](https://github.com/lluchs/openclonk/commit/6caae424e2ba3b6331fedacfb2bba27bd5b33141) for Qt 6 support and plan on doing a release at some point with these updates to keep everything running on newer systems.

**Comment by jcaesar** [comment]
@lluchs This currently building this without the QT editor. (I forgot why. It's a shame, really…) The game itself works fine on the QT 6.

**Comment by lluchs** [comment]
Ah, but without the editor, nothing should link to Qt, right? So there really is no reason to use a Qt runtime then.

Edit: No, I think I misunderstood what you wrote. So it is currently built with the Qt editor, but of course if you use the Qt 6 runtime without my patch, you cannot build with the Qt editor since that requires Qt 5.

**Comment by jcaesar** [comment]
Hm. I thought the Kde runtime provided some additional library that OC needs, but I can't reproduce that right now. It's possible that it was simply set to Kde because I was hoping to build the editor but gave up at some point. :/

I guess I'll be switching to freedesktop then…

**Comment by lluchs** [comment]
But the editor does currently work, doesn't it? I tried installing OpenClonk from flathub earlier and it did have the editor.

**Comment by jcaesar** [comment]
Oh, you're right. I guess I ran the wrong thing when checking… x.x

Whether the editor is built really only depends on whether Qt is found, and it [checks](https://github.com/openclonk/openclonk/blob/db975b4a887883f4413d1ce3181f303d83ee0ab5/CMakeLists.txt#L348-L350) by version. There is no clean way of forcing it to build it or checking whether it was build in either… CMake remains fun for decades.

I guess I'll put it to 5.15-24.08 first, and then we can think about that maintenance release another time (or whether I'll just port that patch to master and then apply it in the flatpak build).

**Comment by jcaesar** [comment]
Will merge in a day or so: https://github.com/flathub/org.openclonk.OpenClonk/pull/9#issuecomment-2751354754
